client/src/components/Voter/VoterContainer.jsx

```diff
@@ -1,11 +1,52 @@
+import useEth from "../../contexts/EthContext/useEth";
+import { useState } from "react";
+
+
 const VoterContainer = () => {
+  const { state: { contract, accounts } } = useEth();
+  const [inputValue, setInputValue] = useState("");
+  const [voter, setVoter] = useState(["", "", ""]);
+  const [error, setError] = useState();
+
+  const handleInputChange = e => {
+    // if (/^\d+$|^$/.test(e.target.value)) {
+    setInputValue(e.target.value);
+    // }
+  };
+
+  const getVoter = async e => {
+    if (e.target.tagName === "INPUT") {
+      return;
+    }
+    if (inputValue === "") {
+      alert("Please enter a value to write.");
+      return;
+    }
+    const newValue = inputValue;
+    await contract.methods.getVoter(newValue).call({ from: accounts[0] }, function (error, result) {
+      if (error) {
+        setError(error.message.match(/revert (.*)/)[1]); // Affiche le message d'erreur du require()
+      } else {
+        setVoter(result);
+      }
+    });
+
+  };
   return (
     <div>
       <h2>Voter</h2>
       <div>
         <label htmlFor="voterAddress">Get a specific voter</label>
-        <input id="voterAddress" type="text" placeholder="Address" />
-        <button>getVoter</button>
+        <input id="voterAddress" type="text" placeholder="Address" value={inputValue}
+          onChange={handleInputChange} accounts={accounts} contract={contract} />
```

Comment:
@rembeuch 
`accounts` et `contract` ne servent à rien dans l'input.
Peut-être un mauvais copier/coller.
```suggestion
          <input id="voterAddress" type="text" value={inputValue} placeholder="Address" 
          onChange={handleInputChange} />
```